src/main/java/com/thebois/models/beings/Colony.java

```diff
@@ -172,4 +192,32 @@ public int numberOf(final ItemType itemType) {
         return inventory.numberOf(itemType);
     }
 
+    /**
+     * Listens to the spawnPawnsEvent in order to know when to add pawns to the colony.
+     *
+     * @param event The published event.
+     */
+    @Subscribe
+    public void onSpawnPawnsEvent(final SpawnPawnsEvent event) {
```

Comment:
Imo, this event is *telling* `Colony` what to do, which isn't its responsibility. This is similar to having a method `spawnPawns` that outside clients can call (just that it's done indirectly through an event instead).

A cleaner way of doing it is by having the event be a `StructurePlacedEvent` that just contains the `StructureType` and `Position` of the structure. This way, the implementation in `World` doesn't need to be specific to any structure type, it can just emit events for all structures whenever they're placed.

And it's also cleaner in here, because then `Colony` will listen using `onStructurePlaced`, and it'll do an easy check of `if(structurePlacedEvent.getType() == StructureType.TOWN_HALL)` and then it'll spawn however many pawns it decides it should for this type of structure. This way, Colony is *deciding* what to do instead of being told.

Reply:
If the logic is moved to house and TownHall, seperating it from AbstractStructure, would you agree with the implementation then?